src/functionalTest/groovy/com/autonomousapps/jvm/projects/BundleProject.groovy

```diff
@@ -0,0 +1,155 @@
+package com.autonomousapps.jvm.projects
+
+import com.autonomousapps.AbstractProject
+import com.autonomousapps.kit.GradleProject
+import com.autonomousapps.kit.Plugin
+import com.autonomousapps.kit.Source
+import com.autonomousapps.kit.SourceType
+import com.autonomousapps.model.Advice
+import com.autonomousapps.model.ProjectAdvice
+
+import static com.autonomousapps.AdviceHelper.*
+import static com.autonomousapps.kit.Dependency.project
+
+final class BundleProject extends AbstractProject {
+
+  final GradleProject gradleProject
+
+  BundleProject() {
+    this.gradleProject = build()
+  }
+
+  private GradleProject build() {
+    def builder = newGradleProjectBuilder()
+    builder.withRootProject {
+      it.withBuildScript { bs ->
+        bs.additions = """
+          dependencyAnalysis {
+            dependencies {
+              bundle('facade') {
+                primary(':entry-point')
```

Comment:
This is what the new API looks like in practice. Note that I haven't added an override to the `bundle()` function, primarily due to limitations in Gradle that make it annoying to create domain objects with custom constructors.